### PR TITLE
Proposal for stat type naming

### PR DESCRIPTION
**Pull request by candy02058912**
Fixes: https://github.com/Automattic/dotcom-forge/issues/4962

## Proposed Changes

* Define Stat Types and granular Sub Types in a constants file
* To use less hardcoded strings and have a single place to check the stat naming

## Testing Instructions

* Make sure the DownloadCSV is still working with the testing instructions in https://github.com/Automattic/wp-calypso/pull/85190

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/stats/hooks/use-should-gate-stats.ts [hunk 1]
```diff
@@ -5,11 +5,21 @@ import isAtomicSite from 'calypso/state/selectors/is-site-wpcom-atomic';
...
-// TODO: define shared variables
-const trafficPaidStats = [ 'statsSearchTerms', 'statsClicks', 'statsReferrers' ];
+const paidStats = [
+	STAT_TYPE_STATS_SEARCH_TERMS,
+	STAT_TYPE_STATS_CLICKS,
+	STAT_TYPE_STATS_REFERRERS,
+];
 
-const featureFlags = [ 'stats/date-control', 'download-csv' ];
+const granularControlForPaidStats = [ STATS_SUBTYPE_DATE_CONTROL, STATS_SUBTYPE_DOWNLOAD_CSV ];
```
Comment (paulopmt1) [inline client/my-sites/stats/hooks/use-should-gate-stats.ts]: Will this `stats/date-control` change also affect [the site.jsx](https://github.com/Automattic/wp-calypso/blob/try/paid-stats-naming/client/my-sites/stats/site.jsx#L219)?

**Review by paulopmt1 — APPROVED** [review]
Nice work @candy02058912!

Tested with:

- Self-hosted: Always showing `Download data as CSV`
- Atomic and simple sites: Without the flag `?flags=stats/paid-wpcom-v2` it downloads as expected. With the flag, it offers the upgrade first.
- After buying the upgrade, it allowed me to download the file.

File: client/my-sites/stats/constants.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export const STAT_TYPE_STATS_TOP_POSTS = 'statsTopPosts';
+export const STAT_TYPE_STATS_VIDEO_PLAYS = 'statsVideoPlays';
+export const STAT_TYPE_STATS_VISITS = 'statsVisits';
+export const STAT_TYPE_TAGS_LIST = 'tagsList';
+export const STAT_TYPE_USERS_LIST = 'usersList';
+export const STAT_TYPE_WPCOM_PLUGINS_LIST = 'wpcomPluginsList';
+
+// Subtypes are for more granular control
+export const STATS_SUBTYPE_DATE_CONTROL = 'StatsDateControl';
+export const STATS_SUBTYPE_DOWNLOAD_CSV = 'StatsDownloadCsv';
```
Comment (lsl) [inline client/my-sites/stats/constants.ts]: I'd just call these features:

```suggestion
export const STATS_FEATURE_DATE_CONTROL = 'StatsDateControl';
export const STATS_FEATURE_DOWNLOAD_CSV = 'StatsDownloadCsv';
```

File: client/my-sites/stats/constants.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+export const STAT_TYPE_STATS_PUBLICIZE = 'statsPublicize';
+export const STAT_TYPE_STATS_REFERRERS = 'statsReferrers';
+export const STAT_TYPE_STATS_SEARCH_TERMS = 'statsSearchTerms';
+export const STAT_TYPE_STATS_STREAK = 'statsStreak';
+export const STAT_TYPE_STATS_SUMMARY = 'statsSummary';
+export const STAT_TYPE_STATS_TAGS = 'statsTags';
+export const STAT_TYPE_STATS_TOP_AUTHORS = 'statsTopAuthors';
+export const STAT_TYPE_STATS_TOP_POSTS = 'statsTopPosts';
+export const STAT_TYPE_STATS_VIDEO_PLAYS = 'statsVideoPlays';
+export const STAT_TYPE_STATS_VISITS = 'statsVisits';
```
Comment (lsl) [inline client/my-sites/stats/constants.ts]: Curious what the source for this prefix is (not this PR), should we drop it in the constants? or is there a third type of stat type?

```suggestion
export const STAT_TYPE_CLICKS = 'statsClicks';
export const STAT_TYPE_COMMENT_FOLLOWERS = 'statsCommentFollowers';
export const STAT_TYPE_COMMENTS = 'statsComments';
export const STAT_TYPE_COUNTRY_VIEWS = 'statsCountryViews';
export const STAT_TYPE_FOLLOWERS = 'statsFollowers';
export const STAT_TYPE_INSIGHTS = 'statsInsights';
export const STAT_TYPE_PUBLICIZE = 'statsPublicize';
export const STAT_TYPE_REFERRERS = 'statsReferrers';
export const STAT_TYPE_SEARCH_TERMS = 'statsSearchTerms';
export const STAT_TYPE_STREAK = 'statsStreak';
export const STAT_TYPE_SUMMARY = 'statsSummary';
export const STAT_TYPE_TAGS = 'statsTags';
export const STAT_TYPE_TOP_AUTHORS = 'statsTopAuthors';
export const STAT_TYPE_TOP_POSTS = 'statsTopPosts';
export const STAT_TYPE_VIDEO_PLAYS = 'statsVideoPlays';
export const STAT_TYPE_VISITS = 'statsVisits';
```

**Review by lsl — APPROVED** [review]

File: client/my-sites/stats/hooks/use-should-gate-stats.ts (hunk 1 above)
Comment (candy02058912) [reply client/my-sites/stats/hooks/use-should-gate-stats.ts]: Nope, that's a different thing, I was using it for a name and wanted to change it later (in this PR) to not cause confusion.

**Pull request merged**